server/cli/launch.py

```diff
@@ -76,24 +77,32 @@ def launch(data, layout, diffexp, title, verbose, debug, obs_names, var_names,
         )
 
     if not verbose:
-        log = logging.getLogger('werkzeug')
+        log = logging.getLogger("werkzeug")
         log.setLevel(logging.ERROR)
 
-    click.echo(f'[cellxgene] Loading data from {basename(data)}, this may take awhile...')
+    click.echo(f"[cellxgene] Loading data from {basename(data)}, this may take awhile...")
 
     from server.app.scanpy_engine.scanpy_engine import ScanpyEngine
 
-    args = {'layout': layout, 'diffexp': diffexp, 'max_category_items': max_category_items,
-            'obs_names': obs_names, 'var_names': var_names}
+    args = {
+        "layout": layout,
+        "diffexp": diffexp,
+        "max_category_items": max_category_items,
+        "obs_names": obs_names,
+        "var_names": var_names
+    }
 
-    app.data = ScanpyEngine(data, args)
+    try:
+        app.data = ScanpyEngine(data, args)
+    except ScanpyFileError as e:
+        raise click.ClickException(f"{e}")
 
     if open_browser:
-        click.echo(f'[cellxgene] Launching! Opening your browser to {cellxgene_url} now.')
+        click.echo(f"[cellxgene] Launching! Opening your browser to {cellxgene_url} now.")
         webbrowser.open(cellxgene_url)
     else:
-        click.echo(f'[cellxgene] Launching! Please go to {cellxgene_url} in your browser.')
+        click.echo(f"[cellxgene] Launching! Please go to {cellxgene_url} in your browser.")
 
-    click.echo('[cellxgene] Type CTRL-C at any time to exit.')
+    click.echo("[cellxgene] Type CTRL-C at any time to exit.")
 
     app.run(host=host, debug=debug, port=port)
```

Comment:
what happened to threading=true?   (is this branch merged with latest master?)

Reply:
Hmm, I thought so. I'll double check

Reply:
rebased